Rename

mikemellor11 opened the pull request:
Hey, i was using this in a project the other day and noticed some strange behaviour when file renames occur. I've added a failing test case and some test cases for added and removed files as well. Basically when a file is renamed the status array and the files array become mixed, i'm not really sure what the desired behaviour is supposed to be, a status code 'R' with the new filename in files?

Here's an example of the commit object when a rename has occurred;

```
{
    "abbrevHash": "b89215b",
    "hash": "b89215b6dd6a2c27e47789bf954e5644e0d0aa78",
    "subject": " files renamed",
    "authorName": "Your Name",
    "status": [
        "R100",
        "1-fileRename",
        "2-file"
    ],
    "files": [
        "1-file",
        "R100",
        "2-fileRename"
    ]
}
```

So here two files have been renamed using

```
mv 1-file 1-fileRename
mv 2-file 2-fileRename
```

As you can see it seems to be mixing the status's and the files at this point, i'm not really sure what the ideal behaviour is supposed to be here, if you could take a look and let me know how its supposed to work, i'm happy to implement the code.

Regards

Mike

**Edit**

Also after seeing the report from CI it seems to output a slightly different object, i'm assuming the CI is set to use an older version of git.

```
{
    "abbrevHash": "4bcd6de",
    "hash": "4bcd6deaee38e3c8393c6c7f3896347a0ce6b591",
    "subject": " files renamed",
    "authorName": "Your Name",
    "status": [
        "A",
        "A"
    ],
    "files": [
        "1-fileRename",
        "2-fileRename"
    ]
}
```

If this is the desired behaviour then it would seem that something has changed in recent versions of git that means renames are reported differently than just Add's. I'm using git 2.9.0

domharrington commented:
Hey @mikemellor11 thanks for your investigation into this issue. I'm not actually too sure what the desired output should be for this either... i've never used the `status` and `files` options...

Can i add you as a contributor to this repo? I've also just remembered that i've not merged in your other changes in #21 so i'm sorry for that too.

domharrington commented:
I've just added you as a contributor :) i'm on a work laptop atm so am not logged in to `npm` to add you to the npm module. If you give me a nudge whenever you wanna make a new version i'll happily oblige.

mikemellor11 commented:
Hey Dom, cheers for adding me. I've solved the issue now, it was due to this line 

```
pos = (index - fields.length) % notOptFields.length
```

This was dependant on their only being a single file and the status code. I've moved some things around and parsed the non optional commits a little differently and it all seems fine now. I've also added a check for git version number so i can change the expected result in the tests accordingly.

I've put these both into the single pull request now so if you're happy then i'll pull them into the main repo from my other pull request.
